.github/workflows/check_new_kernel_configs.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import json
+import argparse
+import sys
+import re
+
+def get_data_from_config(input_file):
+    with open(input_file, 'r') as file:
+        input_config_data = file.read()
+    return input_config_data
+
+def check_kernel(input_file):
+    match = re.search(r'SPECS/(.*?)/', input_file)
+    if match:
+        return match.group(1)
+    else:
+        return None
```

Comment:
Two things:
- I'd think about the method name a bit. Right now it sounds like you're trying to check the actual kernel and it's a bit confusing. If I understand this correctly, you just want to verify if the file is a kernel config.
- We might be overdoing it here a bit with trying to verify if the given file is indeed a kernel config file. Seeing that it's a very simple format, the best we could do is to check if it contains at least one line beginning with `CONFIG_SOMETHING=(y|n|m)`. But even that wouldn't be a really reliable check. With that said, I think this check might be an overkill and I'd just drop it.

Reply:
Nevermind, I see now that it's actually supposed to extract the kernel folder name from the config path. I'm still going to pretend it's not me paying too little attention to the code and blame it on the name of the function instead.:)

Anyway: I'd re-name it to something closer to `read_kernel_name`, `extract_kernel_dir_name`, etc. so the name better reflects what's happening.